src/pal/src/synchmgr/wait.cpp

```diff
@@ -75,6 +75,35 @@ WaitForSingleObject(IN HANDLE hHandle,
 
 /*++
 Function:
+  WaitForSingleObjectPrioritized
+
+Similar to WaitForSingleObject, except uses a LIFO release policy for waiting threads by prioritizing new waiters (registering
+them at the beginning of the wait queue rather than at the end).
+--*/
+DWORD
+PALAPI
+PAL_WaitForSingleObjectPrioritized(IN HANDLE hHandle,
```

Comment:
>PAL_WaitForSingleObjectPrioritized [](start = 0, length = 34)

Is it possible we can use WaitForSingleObject with extra parameter instead of adding one more PAL method?

Reply:
I wanted to keep the WaitForSingleObject signature in sync with the Windows API